src/phasorpy/phasor.py

```diff
@@ -57,50 +56,47 @@ def phasor_calibration(
     modulation0 : array_like, optional
         Radial component of polar coordinates for calibration. Defaults to 1.0.
 
-
     Returns
     -------
-    tuple[NDArray[Any], NDArray[Any]]
-        real:
-            Calibrated real component of phasor coordinates in Cartesian form.
-        imag:
-            Calibrated imaginary component of phasor coordinates in Cartesian
-            form.
+    real, imag:
+        Calibrated real and imaginary components of phasor coordinates.
 
     Examples
     --------
-    >>> real_data = numpy.array([1.0, 2.0, 3.0])
-    >>> imag_data = numpy.array([4.0, 5.0, 6.0])
+    >>> real = numpy.array([1.0, 2.0, 3.0])
+    >>> imag = numpy.array([4.0, 5.0, 6.0])
     >>> phase0 = 0.5
     >>> modulation0 = 2.0
-    >>> calibrated_real, calibrated_imag = phasor_calibration(
-    ...     real_data, imag_data,
+    >>> real_calibrated, imag_calibrated = phasor_calibrate(
+    ...     real, imag,
+    ...     phase0,
+    ...     modulation0
+    ... )
+    >>> real_calibrated
+    array(...)
+    >>> imag_calibrated
+    array(...)
+    >>> phase0 = numpy.array([0.5, 0.2, 0.3])
+    >>> modulation0 = numpy.array([1.5, 2.0, 0.3])
+    >>> real_calibrated, imag_calibrated = phasor_calibrate(
+    ...     real, imag,
     ...     phase0,
     ...     modulation0
     ... )
-    >>> calibrated_real
+    >>> real_calibrated
     array(...)
-    >>> calibrated_imag
+    >>> imag_calibrated
     array(...)
     """
-    phi0, mod0 = numpy.asarray(phase0), numpy.asarray(modulation0)
+    phi0 = numpy.asarray(phase0)
+    mod0 = numpy.asarray(modulation0)
+    re = numpy.asarray(real)
+    im = numpy.asarray(imag)
     if numpy.all(phi0 == 0) and numpy.all(mod0 == 1):
-        return real, imag
-    real = numpy.asarray(real)
-    imag = numpy.asarray(imag)
-    correction_matrix = (
-        numpy.array(
-            (
-                (numpy.cos(phi0), -numpy.sin(phi0)),
-                (numpy.sin(phi0), numpy.cos(phi0)),
-            )
-        )
-        * mod0
-    )
-    real, imag = (
-        correction_matrix.dot(numpy.vstack([real.flatten(), imag.flatten()]))
-    ).reshape((2, *real.shape))
-    return real, imag
+        return re, im
+    re_calibrated = (re * numpy.cos(phi0) - im * numpy.sin(phi0)) * mod0
+    im_calibrated = (re * numpy.sin(phi0) + im * numpy.cos(phi0)) * mod0
+    return re_calibrated, im_calibrated
```

Comment:
Thanks Christoph, It is more efficent that way. Can I ask about the modification in:
` re = numpy.array(real, copy=True)
    im = numpy.array(imag, copy=True)`
    
Is there a reason to always create a copy?

Reply:
A copy is needed because of the in-place modification `im *= cos` and `im_calibrated *= sin`. In fact, it should be verified in the tests that the input data is not modified.

Reply:
You are right, I will add tests to cover this aspect

Reply:
@cgohlke I had to add dtype convertion in the `numpy.array` to float because of casting error with int and float, in the case were real and imag were `int`. I think those cases won't be used too much, but anyway, with this fix it works for all cases:

`    re = numpy.array(real, copy=True, dtype=float)
    im = numpy.array(imag, copy=True, dtype=float)`

Do you think this can alter the process or be less efficent?

Reply:
Yes, we might want to do that for all input coordinates and lifetimes in all functions. One issue is to keep `float32` input as is (to save memory for large images). I see no other way than have a `dtype` input parameter that defaults to `float64` but can be set to `float32`.